elements/dash-checkbox/dash-checkbox.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * A `DashCheckbox` represents a boolean option in a form. The most common type
+ * of checkbox is a dual-type which allows the user to toggle between two
+ * choices -- checked and unchecked.
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    SPACE: 32,
+  };
+
+  /**
+   * The `DashCheckbox` exposes a single `checked` attribute/property for
+   * toggling its state. Changes to the `checked` property will also be
+   * reflected to an `aria-checked` attribute.
+   */
+  class DashCheckbox extends HTMLElement {
+    static get observedAttributes() {
+      return ['checked', 'disabled'];
+    }
+
+    constructor() {
+      super();
+    }
+
+    /**
+     * `connectedCallback` sets the initial role and checks to see if the
+     * user has predefined an `aria-checked` state or `tabindex`.
+     */
+    connectedCallback() {
+      this.setAttribute('role', 'checkbox');
```

Comment:
I guess technically I should check to see if the user has applied their own role and respect that